Sources/MassTransit/Utils/withRunningMassTransitConsumer.swift

```diff
@@ -0,0 +1,33 @@
+import Logging
+import RabbitMq
+
+public func withRunningMassTransitConsumer(
+    using connection: Connection,
+    consumerName: String,
+    routingKey: String = "",
+    configuration: MassTransitConsumerConfiguration = .init(),
+    retryInterval: Duration = MassTransit.defaultRetryInterval,
+    body: @escaping @Sendable (MassTransitConsumer) async throws -> Void
+) async throws {
+    let consumer = MassTransitConsumer(
+        using: connection, queueName: consumerName, exchangeName: consumerName,
+        routingKey: routingKey, configuration: configuration, retryInterval: retryInterval,
+        logger: connection.logger
+    )
+
+    try await withThrowingDiscardingTaskGroup { group in
+        group.addTask {
+            do {
+                try await consumer.run()
+            } catch is CancellationError {
+                // Ignore
+            } catch {
+                throw error
+            }
+        }
+
+        try await body(consumer)
```

Comment:
When consumer.run() returns without any errors, body() will not be cancelled. This may be fine, but I thought I'd mention it.